### ISSUE TITLE
Can't change StyleSheet of Dock Tabs (Qt6 CSS/QSS Windows10)

### ISSUE THREAD

**Issue by LiamC12345** [githubuser0xFFFF/Qt-Advanced-Docking-System#388]
Hello githubuser0xFFFF

Great Docking System - thanks for creating it :)

Basic Docking setup is all done, I've set all the config/feature flags and stuff - that's all fine

I'm just struggling with trying to get my own custom stylesheet working for the Tabs as well as other aspects of the dock
(I.e. change tab fill color, tab text font, border colors etc.)

I have read other posts in the issues forum that you have provided answers to in an attempt to get it working but I still haven't succeeded even after implementing the supposed solutions.

E.g. Post I've read: https://github.com/githubuser0xFFFF/Qt-Advanced-Docking-System/issues/138

Not sure if its a Qt6 thing or my CSS syntax is somehow wrong or whether I'm just simply implementing it incorrectly or not...
-----------------------
But here are some examples of code with different syntax attempts that I have tried (one at a time) in the C++ editor which have not worked:

m_DockManager->setStyleSheet("");
m_DockManager->setStyleSheet("ads--CDockAreaTab {background: rgb(255, 0, 0);}");
m_DockManager->setStyleSheet("ads--CDockAreaTabBar #tabsContainerWidget {background: rgb(255, 0, 0);}");
m_DockManager->setStyleSheet("ads--CDockAreaTab {background-color: rgb(255, 0, 0);}");
m_DockManager->setStyleSheet("ads--CDockAreaTab {color: rgb(255, 0, 0);}");

I Even tried changing individual dock stylesheets directly with the same syntax attempts as above:

dockOne->setStyleSheet("ads--CDockAreaTabBar {background: rgb(255, 0, 0);}");
-----------------------

Nothing has worked thus far
Even editing the default.css document which you provided in the demo hasn't worked either
I've literally kept the same syntax that you have used, not sure why its not working for me

Kind Regards

**Comment by dail8859** [comment]
[This](https://github.com/dail8859/NotepadNext/blob/c566b353358fc3b3c17ec52f66f315d431e24233/src/NotepadNext/stylesheets/npp.css) is the style sheet I use for my application. I use it for both Qt5 and Qt6.

I notice that I am using `CDockWidgetTab` and `CDockAreaTitleBar` however you are using `CDockAreaTab` and `CDockAreaTabBar` so maybe that is the difference? Off the top of my head I really don't know the difference between them. I am by no means an expert :)

Here's how my style sheet looks.

![img](https://raw.githubusercontent.com/dail8859/NotepadNext/c566b353358fc3b3c17ec52f66f315d431e24233/doc/screenshot.png)

**Comment by dail8859** [comment]
Also I'm assuming you are _first_ calling `m_DockManager->setStyleSheet("");` and then calling `setStyleSheet()` again with your custom settings?

**Comment by LiamC12345** [comment]
Thankyou dail8859 for your response

Yeah I tried both first calling m_DockManager->setStyleSheet(""); as well as trying without it to see if anything works.
So far nothing... 

It seems that calling any stylesheet change with m_DockManager just makes the program go straight back
to using the default.css document without applying the new intended setStyleSheet changes.

And also trying to set a stylesheet for a user-made dock doesn't work at all either. 
e.g. customDock->setStyleSheet("some Random Change");
This points to the likelihood that dock stylesheets are global to all docks meaning that different docks can't have different
stylesheets (I hope I'm wrong).

I also did end up trying CDockWidgetTab and CDockAreaTitleBar among others, still to no avail in the C++ editor.
So far trying to change it programatically hasn't worked although I played around a little bit more and finally got some changes
to occur in the default.css by editing the file.

After fiddling around with the default.css stylesheet I've worked out that:
ads--CDockWidgetTab[activeTab="true"] QLabel
//This covers The Tabs Text, font size and color is editable here

ads--CDockWidgetTab[activeTab="true"]
//This covers the Actual Tab itself that the tab label sits in - can change color or even add a resource image here

ads--CDockAreaWidget
//This covers the Dock Bar that the tab sits on, again color can be changed or even resource image can be inserted too

While being able to finally get some changes made via the .css file is great, the limitation of it still is that once these stylesheet
values are set, they cannot later be changed programmatically - unless of course a solution or workaround is eventually found to get custom settings to work when calling setStyleSheet
